_posts/2016-04-05-vision.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+cover: 'assets/images/general-cover-3.jpg'
+title: NectarCommerce Vision
+tags: docs
+subclass: 'post tag-docs'
+categories: 'elixir'
+author: 'Pikender'
+navigation: true
+logo: 'assets/images/nectar-cart.png'
+---
+
+>
+The post belongs to NectarCommerce and Extension Framework Awareness
+>
+1. **NectarCommerce Vision**
+1. Extension Framework Game Plan
+1. Introduction to Writing Macros
+1. Running Multiple Phoenix Apps Together
+1. Ecto Model Extension
+1. Phoenix Router Extension
+1. Phoenix View Extension
+1. Extension Approach Explained
+1. Developer Experience and Workflow developing Favorite Product Extension
+1. Developer Experience and Workflow testing Favorite Product Extension
+
+
+## What will be NectarCommerce
+
+>
+Off-the-shelf Opensource E-commerce application for building online store.
+>
+Provides an Extension Framework to support features not included in core as extensions.
+>
+Strives for un-obstrusive parallel development of NectarCommerce and Extensions
+
+NectarCommerce is committed to provide a ready-to-use e-commerce solution but definition of 100% is different under different business domains.
+
+NectarCommerce aims to solve common use-cases as part of the project and relying on extension framework to tap the rest.
+
+Vision
+======
+
+**E-commerce Framework with an extension methodology which allows easy integration of custom features unique to the domain.**
+
+## NectarCommerce in NutShell
+
+[NectarCommerce](https://github.com/vinsol/nectarcommerce) is an E-commerce Elixir/Phoenix Application which is currently having following features:
+
+- Product Management
+- Order Management
+- Basic Stock Management
+
+Please check out the [demo](https://github.com/vinsol/nectarcommerce#demo) and [README](https://github.com/vinsol/nectarcommerce/blob/master/README.md)
+
+## NectarCommerce and Extensions
+
+Project is in its early stages where NectarCommerce and extensions are residing side-by-side in _Elixir umbrella app_.
+
+Plan is to evolve where Nectar and all other extensions will be mix projects/applications which may have web components via [phoenix](http://www.phoenixframework.org/), which can be downloaded as hex packages.
+
+So, NectarCommerce should provide an [Extension Framework](https://github.com/vinsol/nectarcommerce/pull/47), where custom features as needed per domain can be developed and integrated back to NectarCommerce with minimal effort.
+
+Extensions should seamlessly integrate via Extensions Framework with NectarCommerce such that changes in extensions should not result / demand a change in NectarCommerce Project
+and extensions can incrementally adapt to changes in NectarCommerce for better support, new features and betterments.
+
+NectarCommerce and extensions can be grown and develop in isolations, wherein, Nectar being maintained by one team and extensions probably by many different teams where no one team will own all the extensions and Nectar Project.
```

Comment:
grown and developed in isolation.

NectarCommerce(instead of Nectar) is(instead of being) developed(maintained)  by one team.

NectarCommerce(instead of Nectar Project)
Also, not completely sure on the last sentence.